admin/admin_page_registration.php

```diff
@@ -15,60 +15,57 @@
 if (isset($_POST) && !empty($_POST)) {
   $error = FALSE;
 
-  if (isset($_POST['page_key'])) {
-    $page_key = zen_db_prepare_input($_POST['page_key']);
-  }
-  if (empty($page_key)) {
-    $error = TRUE;
-    $messageStack->add(ERROR_PAGE_KEY_NOT_ENTERED, 'error');
-  } elseif (zen_page_key_exists($page_key)) {
-    $error = TRUE;
-    $messageStack->add(ERROR_PAGE_KEY_ALREADY_EXISTS, 'error');
-  }
+  $keys = [
+            'page_key' => [
+                            'empty' => true,
+                            'elseif' => 'zen_page_key_exists',
+                          ],
+            'language_key' => [
+                                'empty' => true,
+                                'elseif' => 'defined',
+                                'elseifNot' => true,
+                              ],
+            'main_page' => [
+                            'empty' => true,
+                            'elseif' => 'defined',
+                            'elseifNot' => true,
+                           ],
+            'page_params',
+            'menu_key' => [
+                            'empty' => true,
+                          ],
+            'sort_order' => [
+                              'valueType' => 'integer',
+                            ],
+          ];

```

Comment:
Let's stick to PSR-2 for indenting. I find the following much more readable than indenting by "2 chars longer than the last character of the previous line":
```suggestion
    $keys = [
        'page_key' => [
            'empty' => true,
            'elseif' => 'zen_page_key_exists',
        ],
        'language_key' => [
            'empty' => true,
            'elseif' => 'defined',
            'elseifNot' => true,
        ],
        'main_page' => [
            'empty' => true,
            'elseif' => 'defined',
            'elseifNot' => true,
        ],
        'page_params',
        'menu_key' => [
            'empty' => true,
        ],
        'sort_order' => [
            'valueType' => 'integer',
        ],
    ];
```

Reply:
Glad to stick with that, haven't been sure what the expectation has been for such array "lists".
Looking to apply to the bulk of the code, though did also see an operational change that was needed. Both are being committed separately with the PSR-2 change done all in one commit and after the logic correction.